google/cloud/pubsub/internal/flow_controlled_publisher_connection.h

```diff
@@ -59,23 +65,15 @@ class FlowControlledPublisherConnection
 
   void OnPublish(std::size_t message_size);
   bool IsFull() const {
-    return pending_bytes_ >= options_.maximum_pending_bytes() ||
-           pending_messages_ >= options_.maximum_pending_messages();
+    return pending_messages_ > options_.maximum_pending_messages() ||
+           pending_bytes_ > options_.maximum_pending_bytes();
   }
   bool MakesFull(std::size_t message_size) const {
-    if (pending_messages_ + 1 > options_.maximum_pending_messages()) {
-      return true;
-    }
-    return pending_bytes_ + message_size > options_.maximum_pending_bytes();
-  }
-  bool RejectWhenFull() const {
-    return options_.full_publisher_action() ==
-           pubsub::PublisherOptions::kFullPublisherRejects;
-  }
-  bool BlockWhenFull() const {
-    return options_.full_publisher_action() ==
-           pubsub::PublisherOptions::kFullPublisherBlocks;
+    return pending_messages_ + 1 > options_.maximum_pending_messages() ||
+           pending_bytes_ + message_size > options_.maximum_pending_bytes();
```

Comment:
Q. Perhaps the first message should never count as "MakesFull", no matter what its size?  Might that alleviate the "leads to deadlock" issue?

Reply:
Good idea, done.